src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -392,6 +457,30 @@ pub struct VcpuState {
     vcpu_events: kvm_vcpu_events,
     xcrs: kvm_xcrs,
     xsave: kvm_xsave,
+    #[version(start = 2, default_fn = "default_tsc_khz", ser_fn = "ser_tsc")]
+    tsc_khz: Option<u32>,
+}
+
+impl VcpuState {
+    fn default_tsc_khz(_: u16) -> Option<u32> {
+        warn!("CPU TSC freq not found in snapshot");
+        None
+    }
+
+    fn ser_tsc(&mut self, target_version: u16) -> VersionizeResult<()> {
+        if target_version < 2 {
+            warn!("Saving to older snapshot version, TSC freq not included in snapshot.");
+        }
+
+        // Prevent v0.25 to create snapshots of uVMs running on non-invariant TSC host.
+        if self.tsc_khz.is_none() {
```

Comment:
We dont't allow this but we do load snaps from v0.24 (without TSC) given the CPUID model information matches. What happens if we drop this check ?

Reply:
Depends on whether you want to load the TSC-less snapshot on the same CPU model or not. I guess this ties in to the previous comment.

Reply:
I think we should have a consistent behavior both for load and save wrt older snapshot versions.

Reply:
Dropped the check, now the failure path is on snapshot restore. We still warn users about this with a log.